examples/src/examples/user-interface/world-to-screen.tsx

```diff
@@ -96,30 +96,80 @@ class WorldScreenExample extends Example {
+            name.addComponent("element", {
                 pivot: new pc.Vec2(0.5, 0.5),
-                anchor: new pc.Vec4(0.5, 0.5, 0.5, 0.5),
+                anchor: new pc.Vec4(0, 0.4, 1, 1),
+                margin: new pc.Vec4(0, 0, 0, 0),
                 fontAsset: assets.font.id,
                 fontSize: 20,
                 text: `Player ${id}`,
                 useInput: true,
                 type: pc.ELEMENTTYPE_TEXT
             });
-            playerText.addComponent("button", {
-                imageEntity: playerText
+            name.addComponent("button", {
+                imageEntity: name
             });
-            playerText.button.on('click', function (e) {
+            name.button.on('click', function (e) {
                 const color = new pc.Color(Math.random(), Math.random(), Math.random());
-                playerText.element.color = color;
+                name.element.color = color;
                 entity.render.material.setParameter("material_diffuse", [color.r, color.g, color.b]);
             });
-            screen.addChild(playerText);
+            playerInfo.addChild(name);
+
+            const healthBar = new pc.Entity();
+            healthBar.addComponent("element", {
+                pivot: new pc.Vec2(0.5, 0),
+                anchor: new pc.Vec4(0, 0, 1, 0.4),
+                margin: new pc.Vec4(0, 0, 0, 0),
+                color: new pc.Color(0.2, 0.6, 0.2, 1),
+                opacity: 1,
+                type: pc.ELEMENTTYPE_IMAGE
+            });
+            playerInfo.addChild(healthBar);
 
             // update the player text's position to always hover the player
             app.on("update", function () {
-                const playerPosition = entity.getPosition();
-                playerText.setPosition(playerPosition.x, playerPosition.y + 0.6, playerPosition.z);
+                // get the desired world position
+                const worldPosition = entity.getPosition();
+                worldPosition.y += 0.6; // slightly above the player's head
+
+                // convert to screen position
+                const screenPos = camera.camera.worldToScreen(worldPosition);
+
+                if (screenPos.z > 0) {
+                    // if world position is in front of the camera, show it
+                    playerInfo.enabled = true;
+
+                    // take pixel ratio into account
+                    const pixelRatio = app.graphicsDevice.maxPixelRatio;
+                    screenPos.x *= pixelRatio;
+                    screenPos.y *= pixelRatio;
+
+                    // account for screen scalng
+                    // @ts-ignore engine-tsd
+                    const scale = screen.screen.scale;
+
+                    // invert the y position
+                    screenPos.y = screen.screen.resolution.y - screenPos.y;
+
+                    // set the UI position
+                    playerInfo.setLocalPosition(screenPos.x / scale, screenPos.y / scale, 0);
```

Comment:
it'd be good to convert some / all of this to a function, so if people need to do something similar, they can just grab it from here.